packages/loader/container-loader/src/container.ts

```diff
@@ -838,6 +839,12 @@ export class Container extends EventEmitterWithErrorHandling<IContainerEvents> i
             if (!this.closed) {
                 this.resumeInternal({ fetchOpsFromStorage: false, reason: "createDetached" });
             }
+        } catch(error) {
+            if (!canRetryOnError(error)) {
+                this.logger.sendErrorEvent({ eventName: "AttachFail", attachState: this._attachState }, error);
```

Comment:
this will double log the error, as close will log it to. i don't think the pre-log is necessary

Reply:
no. i don't think it should be public. i just don't really like that if attach fails all the data is gone, and there isn't anything else that can be done

Reply:
I think this is fine for now. I've been thinking about how to support incremental modifications detached containers, which is something i think we'll need to solve eventually. hopefully in solving that we can make attach safer to retry, so we don't loose all the data on an attach failure.